src/components/__tests__/App-test.tsx

```diff
@@ -4,24 +4,55 @@ import * as React from "react";
 import * as Enzyme from "enzyme";
 import App from "../App";
 import Header from "../reusables/Header";
+import LogInFrom from "../reusables/LogInForm";
 import LibrariesListContainer from "../LibrariesListContainer";
+import { MemoryRouter } from "react-router-dom";
+import buildStore from "../../store";
+import ContextProvider from "../ContextProvider";
 
 describe("App", () => {
-  let wrapper: Enzyme.ShallowWrapper<any, {}>;
+  let wrapper: Enzyme.ShallowWrapper<{}, {}>;
+  let context;
+  let store;
+
   beforeEach(() => {
-    wrapper = Enzyme.shallow(
-      <App />
+    store = buildStore();
+    context = { store, username: "" };
+    wrapper = Enzyme.mount(
+      <ContextProvider {...context}>
+        <MemoryRouter initialEntries={["/admin"]}>
```

Comment:
`MemoryRouter` is useful for tests (and react native). It's creating a router and declaring the path that we want.